benchmarks/bench_gap_encoder.py

```diff
@@ -0,0 +1,1332 @@
+        batch_size=batch_size,
+        hashing=hashing,
+        n_components=n_components,
+        ngram_range=ngram_range,
+        analyzer=analyzer,
+        init=init,
+        tol=tol,
+        min_iter=min_iter,
+        max_iter=max_iter,
+        gamma_shape_prior=gamma_shape_prior,
+        gamma_scale_prior=gamma_scale_prior,
+        rho=rho,
+        rescale_rho=rescale_rho,
+        rescale_W=rescale_W,
+        max_iter_e_step=max_iter_e_step,
+    )
+
+    start_time = perf_counter()
+    gap.fit(X_train)
+    end_time = perf_counter()
+    score_train = gap.score(X_train)
+    score_test = gap.score(X_test)
+
+    # evaluate the accuracy using the encoding
+    X_train_encoded = gap.transform(X_train)
+    X_test_encoded = gap.transform(X_test)
+    clf = LogisticRegression(max_iter=1000)
+    clf.fit(X_train_encoded, y_train)
+    roc_auc_logreg_train = roc_auc_score(
+        y_train, clf.predict_proba(X_train_encoded)[:, 1]
+    )
+    roc_auc_logreg_test = roc_auc_score(y_test, clf.predict_proba(X_test_encoded)[:, 1])
+    balanced_accuracy_logreg_train = balanced_accuracy_score(
+        y_train, clf.predict(X_train_encoded)
+    )
+    balanced_accuracy_logreg_test = balanced_accuracy_score(
+        y_test, clf.predict(X_test_encoded)
+    )
+
+    clf = HistGradientBoostingClassifier()
+    clf.fit(X_train_encoded, y_train)
+    roc_auc_hgb_train = roc_auc_score(y_train, clf.predict_proba(X_train_encoded)[:, 1])
+    roc_auc_hgb_test = roc_auc_score(y_test, clf.predict_proba(X_test_encoded)[:, 1])
+    balanced_accuracy_hgb_train = balanced_accuracy_score(
+        y_train, clf.predict(X_train_encoded)
+    )
+    balanced_accuracy_hgb_test = balanced_accuracy_score(
+        y_test, clf.predict(X_test_encoded)
+    )
+
+    res_dic = {
+        "time_fit": end_time - start_time,
+        "score_train": score_train,
+        "score_test": score_test,
+        "roc_auc_logreg_train": roc_auc_logreg_train,
+        "roc_auc_logreg_test": roc_auc_logreg_test,
+        "roc_auc_hgb_train": roc_auc_hgb_train,
+        "roc_auc_hgb_test": roc_auc_hgb_test,
+        "balanced_accuracy_logreg_train": balanced_accuracy_logreg_train,
+        "balanced_accuracy_logreg_test": balanced_accuracy_logreg_test,
+        "balanced_accuracy_hgb_train": balanced_accuracy_hgb_train,
+        "balanced_accuracy_hgb_test": balanced_accuracy_hgb_test,
+        "train_size": X_train.shape[0],
+        "test_size": X_test.shape[0],
+    }
+
+    print("Done")
+    print(res_dic)
+
+    return res_dic
+
+
+def plot(df: pd.DataFrame):
+    print("Not implemented yet")
+
+
+if __name__ == "__main__":
+    from argparse import ArgumentParser
+
+    _args = ArgumentParser(
+        description="Benchmark for the batch feature of the MinHashEncoder.",
+        parents=[default_parser],
+    ).parse_args()
+
+    if _args.run:
+        print("Running benchmark")
+        ds = fetch_traffic_violations()
+        df = benchmark(ds)
```

Comment:
Passing both a `parametrize` in `monitor` and the dataset in the call is not supported (check `monitor`'s docstring).
Just load the dataset inside the benchmark function, it is loaded from disk after being downloaded during the first fetch call.